cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultPluginContext.java

```diff
@@ -64,6 +66,24 @@ public PluginProperties getPluginProperties(String pluginId) {
   }
 
   @Override
+  public PluginProperties getPluginProperties(String pluginId, MacroEvaluator evaluator) {
+    Plugin plugin = getPlugin(pluginId);
```

Comment:
Is this copied from the `PluginInstantiate.substituteMacros` method?? Maybe we should just expose that method and call it from here?